www/js/splash/notifScheduler.ts

```diff
@@ -0,0 +1,286 @@
+   This is only called once when the user first joins the study
+*/
+const initReminderPrefs = (reminderSchemes) => {
+    // randomly assign from the schemes listed in config
+    const schemes = Object.keys(reminderSchemes);
+    const randAssignment = schemes[Math.floor(Math.random() * schemes.length)];
+    const todayDate = DateTime.local().toFormat('yyyy-MM-dd');
+    const defaultTime = reminderSchemes[randAssignment]?.defaultTime || '12:00';
+    return {
+        reminder_assignment: randAssignment,
+        reminder_join_date: todayDate,
+        reminder_time_of_day: defaultTime,
+    };
+}
+
+/* EXAMPLE VALUES - present in user profile object
+    reminder_assignment: 'passive',
+    reminder_join_date: '2023-05-09',
+    reminder_time_of_day: '21:00',
+*/
+// interface ReminderPrefs {
+//     reminder_assignment: string;
+//     reminder_join_date: string;
+//     reminder_time_of_day: string;
+// }
+
+const getReminderPrefs = async (reminderSchemes): Promise<any> => {
+    const user = await getUser();
+    if (user?.reminder_assignment &&
+        user?.reminder_join_date &&
+        user?.reminder_time_of_day) {
+        console.log("User already has reminder prefs, returning them", user)
+        return user;
+    }
+    // if no prefs, user just joined, so initialize them
+    console.log("User just joined, Initializing reminder prefs")
+    const initPrefs = initReminderPrefs(reminderSchemes);
+    console.log("Initialized reminder prefs: ", initPrefs);
+    await setReminderPrefs(initPrefs, reminderSchemes);
+    return { ...user, ...initPrefs }; // user profile + the new prefs
+}
+const setReminderPrefs = async (newPrefs, reminderSchemes) => {
+    await updateUser(newPrefs)
+    const updatePromise = new Promise<void>((resolve, reject) => {
+        //enforcing update before moving on
+        update(reminderSchemes).then(() => {
+            resolve();
+        });
+    });
+    // record the new prefs in client stats
+    getReminderPrefs(reminderSchemes).then((prefs) => {
+        // extract only the relevant fields from the prefs,
+        // and add as a reading to client stats
+        const { reminder_assignment,
+            reminder_join_date,
+            reminder_time_of_day} = prefs;
+        addStatReading(statKeys.REMINDER_PREFS, {
+            reminder_assignment,
+            reminder_join_date,
+            reminder_time_of_day
+        }).then(logDebug("Added reminder prefs to client stats"));
+    });
+    return updatePromise;
+}
+
+export function useSchedulerHelper() {
+    const appConfig = useAppConfig();
+    const [reminderSchemes, setReminderSchemes] = useState();
+
+    useEffect(() => {
+        if (!appConfig) {
+            logDebug("No reminder schemes found in config, not scheduling notifications");
+            return;
+        }
+        setReminderSchemes(appConfig.reminderSchemes);
+    }, [appConfig]);
+
+    //setUpActions();
+    update(reminderSchemes);
```

Comment:
If you remove the React hook, you'll need to do this `update` call somewhere else. You can do it in `ProfileSettings` somewhere in the `whenReady` function.